src/main/java/de/torui/coflsky/CoflSkyCommand.java

```diff
@@ -20,201 +22,245 @@
+            } else {
+                ListHelp(sender);
+            }
+        }).start();
+    }
+
+    private void HandleReset() {
+        CoflSky.Wrapper.SendMessage(new Command<String>(CommandType.Reset, ""));
+        CoflSky.Wrapper.stop();
+        Minecraft.getMinecraft().thePlayer.addChatMessage(new ChatComponentText("Stopping Connection to CoflNet"));
+        CoflSessionManager.DeleteAllCoflSessions();
+        Minecraft.getMinecraft().thePlayer.addChatMessage(new ChatComponentText("Deleting CoflNet sessions..."));
+        if (CoflSky.Wrapper.startConnection())
+            Minecraft.getMinecraft().thePlayer.addChatMessage(new ChatComponentText("Started the Connection to CoflNet"));
+    }
+
+    public String StatusMessage() {
+
+        String vendor = System.getProperty("java.vm.vendor");
+        String name = System.getProperty("java.vm.name");
+        String version = System.getProperty("java.version");
+        String detailedVersion = System.getProperty("java.vm.version");
+
+        String status = vendor + " " + name + " " + version + " " + detailedVersion + "|Connection = " + (CoflSky.Wrapper != null ? CoflSky.Wrapper.GetStatus() : "UNINITIALIZED_WRAPPER");
+        try {
+            status += "  uri=" + CoflSky.Wrapper.socket.uri.toString();
+        } catch (NullPointerException npe) {
+        }
+
+
+        try {
+            CoflSession session = CoflSessionManager.GetCoflSession(PlayerDataProvider.getUsername());
+            String sessionString = CoflSessionManager.gson.toJson(session);
+            status += "  session=" + sessionString;
+        } catch (IOException e) {
+        }
+
+        return status;
+    }
+
+    public void SendCommandToServer(String[] args, ICommandSender sender) {
+        String command = String.join(" ", Arrays.copyOfRange(args, 1, args.length));
+
+        //JsonStringCommand sc = new JsonStringCommand(args[0], WSClient.gson.toJson(command));
+        RawCommand rc = new RawCommand(args[0], WSClient.gson.toJson(command));
+        if (CoflSky.Wrapper.isRunning) {
+            CoflSky.Wrapper.SendMessage(rc);
+        } else {
+            SendAfterStart(sender, rc);
+        }
+    }
+
+    private static synchronized void SendAfterStart(ICommandSender sender, RawCommand rc) {
+        sender.addChatMessage(new ChatComponentText("CoflSky wasn't active.").setChatStyle(new ChatStyle().setColor(EnumChatFormatting.RED)));
+        //CoflSky.Wrapper.stop();
+        CoflSky.Wrapper.startConnection();
+        CoflSky.Wrapper.SendMessage(rc);
+    }
+
+    public void ListHelp(ICommandSender sender) {
+        sender.addChatMessage(new ChatComponentText(HelpText));
+        sender.addChatMessage(new ChatComponentText(QueryServerCommands.QueryCommands()));
+    }
+
+    public void CallbackCommand(String[] args) {
+
+        String command = String.join(" ", Arrays.copyOfRange(args, 1, args.length));
+        System.out.println("CallbackData: " + command);
+        //new Thread(()->{
```

Comment:
All commands have to be offloaded to another thread in order to not block the rendering thread

Reply:
I dont really get what I am supposed to change here? Remove the comment?
I didnt add a new command, just a new parameter. And the entire `processCommand` method is in a new Thread (see CoflSkyCommand.java:67).

This "new Thread" was commented out by Rinke in 2021